samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImplGraph.java

```diff
@@ -217,9 +221,13 @@ OperatorImpl createOperatorImpl(OperatorSpec prevOperatorSpec, OperatorSpec oper
     } else if (operatorSpec instanceof SinkOperatorSpec) {
       return new SinkOperatorImpl((SinkOperatorSpec) operatorSpec, config, context);
     } else if (operatorSpec instanceof OutputOperatorSpec) {
-      return new OutputOperatorImpl((OutputOperatorSpec) operatorSpec);
+      String streamId = ((OutputOperatorSpec) operatorSpec).getOutputStream().getStreamId();
+      SystemStream systemStream = util.toSystemStream(streamId);
+      return new OutputOperatorImpl((OutputOperatorSpec) operatorSpec, systemStream);
     } else if (operatorSpec instanceof PartitionByOperatorSpec) {
-      return new PartitionByOperatorImpl((PartitionByOperatorSpec) operatorSpec, config, context);
+      String streamId = ((PartitionByOperatorSpec) operatorSpec).getOutputStream().getStreamId();
```

Comment:
Just to double check:
Does the planner rewrite the config to add intermediate stream info before this runs?

Reply:
Yeah, it does so in JobNode#generateConfig():
`inEdges.stream().filter(StreamEdge::isIntermediate).forEach(edge -> configs.putAll(edge.generateConfig()));`

In StreamEdge#generateConfig:
```
if (isIntermediate()) {
   config.put(String.format(StreamConfig.IS_INTERMEDIATE_FOR_STREAM_ID(), spec.getId()), "true");
   ...
}
```